From 5e1784e3f8aaa843d032f1c3504efdfbd3f68ce8 Mon Sep 17 00:00:00 2001
From: Steven Barrett <steven@liquorix.net>
Date: Tue, 5 Nov 2019 15:49:24 -0600
Subject: [PATCH 2/2] mm: Raise hugepage compaction effort to 60

Use the same value to keep external fragmentation under control as
tested by Nitin Gupta.
---
 mm/compaction.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mm/compaction.c b/mm/compaction.c
index 9a5f85e8b7ac..89f2c1f36136 100644
--- a/mm/compaction.c
+++ b/mm/compaction.c
@@ -2599,7 +2599,7 @@ static void __init compaction_init(void)
 		struct compaction_state *c = &compaction_states[node];
 
 		c->node_id = node;
-		c->hpage_compaction_effort = 0;
+		c->hpage_compaction_effort = 60;
 		snprintf(c->name, COMPACTION_STATE_NAME_LEN, "node-%d", node);
 	}
 }
-- 
2.24.0

